bot/cogs/snakes.py

```diff
@@ -25,23 +47,183 @@ def __init__(self, bot: AutoShardedBot):
 
-        :param name: Optional, the name of the snake to get information for - omit for a random snake
+        :param name: The name of the snake to get information for - omit for a random snake
         :return: A dict containing information on a snake
         """
+        snake_info = {}
+        # python (programming language) pageid = 23862
+        URL = "https://en.wikipedia.org/w/api.php?"
+        ACTION = "action=query"
+        LIST = "list=search"
+        SRSEARCH = "srsearch="
+        UTF8 = "utf8="
+        SRLIMIT = "srlimit=1"
+        FORMAT = "format=json"
+        PROP = "prop=extracts|images|info"
+        EXLIMIT = "exlimit=max"
+        EXPLAINTEXT = "explaintext"
+        INPROP = "inprop=url"
 
-    @command()
-    async def get(self, ctx: Context, name: str = None):
-        """
-        Go online and fetch information about a snake
+        PAGE_ID_URL = f"{URL}{FORMAT}&{ACTION}&{LIST}&{SRSEARCH}{name}&{UTF8}&{SRLIMIT}"
+
+        async with aiohttp.ClientSession() as session:
+            j = await self.fetch(session, PAGE_ID_URL)
+            # wikipedia does have a error page
+            try:
+                PAGEID = j["query"]["search"][0]["pageid"]
+            except KeyError:
+                PAGEID = 41118
+            PAGEIDS = f"pageids={PAGEID}"
+
+            snake_page = f"{URL}{FORMAT}&{ACTION}&{PROP}&{EXLIMIT}&{EXPLAINTEXT}&{INPROP}&{PAGEIDS}"
 
-        This should make use of your `get_snek` method, using it to get information about a snake. This information
-        should be sent back to Discord in an embed.
+            j = await self.fetch(session, snake_page)
+            # constructing dict - handle exceptions later
+            try:
+                snake_info["title"] = j["query"]["pages"][f"{PAGEID}"]["title"]
+                snake_info["extract"] = j["query"]["pages"][f"{PAGEID}"]["extract"]
+                snake_info["images"] = j["query"]["pages"][f"{PAGEID}"]["images"]
+                snake_info["fullurl"] = j["query"]["pages"][f"{PAGEID}"]["fullurl"]
+                snake_info["pageid"] = j["query"]["pages"][f"{PAGEID}"]["pageid"]
+            except KeyError:
+                snake_info["error"] = True
+            if snake_info["images"]:
+                i_url = 'https://commons.wikimedia.org/wiki/Special:FilePath/'
+                image_list = []
+                map_list = []
+                thumb_list = []
+                banned = ['Commons-logo.svg',
+                          'Red%20Pencil%20Icon.png',
+                          'distribution',
+                          'The%20Death%20of%20Cleopatra%20arthur.jpg',
+                          'Head%20of%20holotype',
+                          'locator',
+                          'Woma.png',
+                          '-map.',
+                          '.svg',
+                          'ange.',
+                          'Adder%20(PSF).png'
+                          ]
```

Comment:
please be consistent about this style choice. Later on you're using the convention that looks like this:

```py
var = (
    "contents"
)
```

This is the style used in all of our open source projects, so we would prefer if you used this style here as well.